HeartbeatLogging/Sources/HeartbeatStorage.swift

```diff
@@ -48,20 +50,20 @@ final class HeartbeatStorage {
   func flush() -> HeartbeatInfo? {
     queue.sync {
       let flushed = try? load(from: storage)
-      try? save(nil, to: storage)
-      return flushed
+      let saveResult = Result { try save(nil, to: storage) }
+      return try? saveResult.map { flushed }.get()
```

Comment:
This line looked a bit counterintuitive to me. But after I read [Result](https://developer.apple.com/documentation/swift/result) docs it makes more sense. Though I still don't like `map` and `get` because they force the code reader to read (or remember) the docs to understand the code. Just complaining, no actions required 😛.

Reply:
Sorry about that– I want it to be clear! Reworked in 63b041c